File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        long long a, b;
        for(auto t: tokens) {
            if(t == "+" or t == "-" or t == "*" or t == "/") {
                b = st.top(); st.pop();
                a = st.top(); st.pop();
                if(t == "+") a = a + b;
                if(t == "-") a = a - b;
                if(t == "*") a = a * b;
                if(t == "/") a = a / b;
                st.push(a);
            }
            else st.push(stoi(t));
        }
        return st.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        i = 0
        while i < len(tokens):
            operand = tokens[i]
            if operand == '+':
                tokens[i-2] = int(tokens[i-2]) + int(tokens[i-1])
                tokens.pop(i)
                tokens.pop(i-1)
                i -= 1
            elif operand == '-':
                tokens[i-2] = int(tokens[i-2]) - int(tokens[i-1])
                tokens.pop(i)
                tokens.pop(i-1)
                i -= 1
            elif operand == '*':
                tokens[i-2] = int(tokens[i-2]) * int(tokens[i-1])
                tokens.pop(i)
                tokens.pop(i-1)
                i -= 1
            elif operand == '/':
                tokens[i-2] = int(int(tokens[i-2]) / int(tokens[i-1]))
                tokens.pop(i)
                tokens.pop(i-1)
                i -= 1
            else:
                i += 1
        return int(tokens[-1])